src/frontend/src/store/chatStore.ts

```diff
@@ -0,0 +1,83 @@
+
+import { create } from 'zustand';
+import { devtools } from 'zustand/middleware';
+import type { ChatStatus, StreamEvent } from '../types/chat';
+
+export interface ChatUIState {
+    status: ChatStatus;
+    events: StreamEvent[];
+}  
+
+const CHAT_ACTION_TYPES = {
+  startStreaming: 'chat/startStreaming',
+  addEvent: 'chat/addEvent',
+  finishStreaming: 'chat/finishStreaming',
+  setError: 'chat/setError',
+  reset: 'chat/reset',
+  anonymous: 'chat/anonymous',
+} as const;
+
+
+const createInitialState = (): Omit<
+  ChatUIState,
+  'startStreaming' | 'addEventFromPayload' | 'finishStreaming' | 'setError' | 'reset'
```

Comment:
```suggestion
  'startStreaming' | 'addEvent' | 'finishStreaming' | 'setError' | 'reset'
```
...unless you'd rather update the `addEvent` option in `CHAT_ACTION_TYPES` to `addEventFromPayload`.